### PR TITLE
[firewall] ingress filtering

### PR DESCRIPTION
**Pull request by superwhd**
Re-sending the PR that implements firewall. It's the same as https://github.com/openthread/openthread/pull/7043. The previous PR was reverted due to flaky CI.

The following is copied from https://github.com/openthread/openthread/pull/7043.

-----

This PR implements OTBR firewall. It should be merged together with https://github.com/openthread/ot-br-posix/pull/1063. This implementation focuses on ingress filtering. We may also introduce egress filtering when necessary.

For security purpose, there are some packet forwarding rules to follow, which were originally introduced in the spec.

Inbound packets initiated with On-Link addresses source (OMR and mesh local prefix based addresses) should be blocked.
Inbound unicast packets whose destination address is not OMR address or DUA should be blocked.
Inbound unicast packets whose source address or destination address is link-local should be blocked. Note that we don’t need to explicitly add rules for link-local addresses since this should already be handled by the kernel.
These rules can be easily implemented by iptables and ipset.

Before otbr-agent starts, there is a script creating the iptables rules. The rules themselves are constant so we don't need to change them dynamically.
During the runtime of otbr-agent, otbr-agent updates ipsets accordingly whenever there's a change of on-link prefixes.

Depends-On: openthread/ot-br-posix#1063

### CODE REVIEW

File: src/posix/platform/utils.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#include "common/code_utils.hpp"
+#include "common/logging.hpp"
+
+namespace ot {
+namespace Posix {
+
+enum
+{
+    kSystemCommandMaxLength = 1024, ///< Max length of a system call command.
+    kOutputBufferSize       = 256,  ///< Buffer size of command output.
```
Comment (simonlingoogle) [inline src/posix/platform/utils.cpp:49]:
```suggestion
    kOutputBufferSize       = 1024,  ///< Buffer size of command output.
```

256 seems short to me.

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

Thanks.

File: src/posix/platform/firewall.hpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+#include "posix/platform/openthread-posix-config.h"
+
+#include "common/logging.hpp"
+#include "openthread/thread.h"
+
+namespace ot {
+namespace Posix {
+
+#if OPENTHREAD_POSIX_CONFIG_FIREWALL_ENABLE
```
Comment (jwhui) [inline src/posix/platform/firewall.hpp:40]: nit: how about moving this to just after `#include "posix/platform/openthread-posix-config.h"` to minimize include overhead?

File: src/posix/platform/utils.hpp [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+namespace ot {
+namespace Posix {
+
+/**
+ * This method formats a system command to execute.
+ *
+ * @param[in] aFormat  A pointer to the format string.
+ * @param[in] ...      Arguments for the format specification.
+ *
+ * @returns The command exit code.
```
Comment (jwhui) [inline src/posix/platform/utils.hpp:43]: nit: the exit code is an `otError`? Can we document the `otError` values?

**Review by jwhui — APPROVED** [review]

**Pull request merged**